### PR TITLE
Correcting a bug with large numbers and improving some macros

### PR DESCRIPTION
**Pull request by GuillermoCasas**
There was a bug by which very large number of particles were not processed correctly due to the limited range on ints. I also made the macros safer and  optimized some of them a bit.

### CODE REVIEW

File: applications/DEM_application/custom_utilities/inlet.cpp [hunk 1]
```diff
@@ -345,6 +345,7 @@ namespace Kratos {
             int total_mesh_size_accross_mpi_processes = mesh_size_elements; //temporary value until reduction is done
             r_modelpart.GetCommunicator().SumAll(total_mesh_size_accross_mpi_processes);
             const double this_mpi_process_portion_of_inlet_mesh = (double) mesh_size_elements / (double) total_mesh_size_accross_mpi_processes;
+            assert(mp[INLET_NUMBER_OF_PARTICLES] >= 0);
```
Comment (maceligueta) [inline applications/DEM_application/custom_utilities/inlet.cpp]: Are you sure you want to use assert? It is not used anywhere in Kratos

Comment (GuillermoCasas) [reply applications/DEM_application/custom_utilities/inlet.cpp]: Ok, I replaced it for something better.

File: applications/DEM_application/custom_utilities/inlet.cpp [hunk 2]
```diff
@@ -662,5 +661,18 @@ namespace Kratos {
...
+    double DEM_Inlet::GetInputNumberOfParticles(const ModelPart& mp)
+    {
+        double num_part_surface_time = mp[INLET_NUMBER_OF_PARTICLES];
+
+        if (num_part_surface_time >= 0){
+           return num_part_surface_time;
+        }
+
+        else {
+            KRATOS_THROW_ERROR(std::runtime_error, "The value of the Model Part variable INLET_NUMBER_OF_PARTICLES is not a positive int: ", num_part_surface_time);
```
Comment (maceligueta) [inline applications/DEM_application/custom_utilities/inlet.cpp]: I have seen in other PRs that the recommended sintax here is KRATOS_ERROR<<"String" ...
KRATOS_THROW_ERROR is the old style, we have to change some of them.

**Review by maceligueta — CHANGES_REQUESTED** [review]
Some undesired files were uploaded in the last commit!
TimesPartialRelease, console_output, ...

**Review by maceligueta — APPROVED** [review]

**Comment by maceligueta** [comment]
Should we delete this branch?

**Comment by GuillermoCasas** [comment]
No, since I am still working on it.

**Pull request merged**